content/dev/languages/creating_a_language_pack.md

```diff
@@ -0,0 +1,291 @@
+
+           - modules
+
+               - order_total
+
+                   - ot_cod_fee.php
+
+                   - more php files
+
+                   - classic
+
+                       - empty.txt
+
+               - payment
+
+                   - authorizenet.php
+
+                   - more php files
+
+                   - classic
+
+                       - empty.txt
+
+               - shipping
+
+                   - flat.php
+
+                   - more php files
+
+                   - classic
+
+                       - empty.txt
+
+           - responsive_classic
+
+               - icon_names.php
+
+    - templates
+
+        - responsive_classic
+
+            - buttons
+
+                - english
+
+                    - button_update_cart.png
+
+        - template_default
+
+            - buttons
+
+                - english
+
+                    - button_add_address.gif
+
+                    - more gif files
+
+
+
+**Note** You should have this complete directory structure with `english` replaced by `YOURLANGUAGE`
+
+## Modifying the Language Files
+
+Having copied all the English files you should modify the define statements within the files to your chosen language.
+
+e.g. for a Welsh language translation modifying english.php
+
+``` 
+// Define the name of your Gift Certificate as Gift Voucher, Gift Certificate, Zen Cart Dollars, etc. here for use through out the shop
+  define('TEXT_GV_NAME','Tystysgrif Rhodd');
+  define('TEXT_GV_NAMES''Tystysgrifau Rhodd');
+
+// used for redeem code, redemption code, or redemption id
+  define('TEXT_GV_REDEEM','Cod Adbrynu');
+
+// text for gender
+  define('MALE', 'Br.');
+  define('FEMALE', 'Bns.');
+```
+
+## Creating SQL patch for Admin
+
+You can extract a full set of the admin descriptions from the configuration table. Using any database management tool.
+
+`SELECT configuration_key, configuration_description FROM configuration;`
+
+You will get a file containing:
+```
+STORE_COUNTRY	The country my store is located in <br /><br /><strong>Note: Please remember to update the store zone.</strong>
+STORE_NAME	The name of my store
+STORE_OWNER	The name of my store owner
+STORE_ZONE	The zone my store is located in
+...
+
+```
+
+You then need to create a update file to modify the descriptions on the database.
+
+E.g. To change the descriptions above from English to the Welsh you could use:
+
+```
+UPDATE configuration SET configuration_description = 'Y wlad y mae fy siop wedi'i lleoli yn <br /> <br /> <strong> Nodyn: Cofiwch ddiweddaru'r parth storfa. </strong>' WHERE configuration_key = 'STORE_COUNTRY';
+UPDATE configuration SET configuration_description = 'Enw fy siop' WHERE configuration_key = 'STORE_NAME';
+UPDATE configuration SET configuration_description = 'Enw perchennog fy siop' WHERE configuration_key = 'STORE_OWNER';
+UPDATE configuration SET configuration_description = 'Mae\'r parth y mae fy siop wedi\'i leoli ynddo' WHERE configuration_key = 'STORE_ZONE';

```

Comment:
I want to scream "NOOOOOOOO!" here.

Tell them to create language-defines that override each of the configuration_keys, for their own language. Don't change the database contents, because upgrades are going to wipe out those changes for any keys that get the title/description updated from time to time.

Reply:
Here's how:
https://github.com/zencart/zencart/blob/6e1e59f8ab7e552f4232e1ce8ff02f9c0a2706f7/admin/configuration.php#L139-L142